src/hooks/use-stores.ts

```diff
@@ -70,6 +72,22 @@ export function useDocumentOrCurriculumMetadata(key?: string): IDocumentMetadata
   }, [documentMetadata, key]);
 }
 
+export function useCurriculumContent(key?: string):  DocumentContentModelType | undefined {
+  const curriculumContentFromPath = useCurriculumContentFromPath(key); 
+  const curriculumContentFromKey = useDocumentFromStore(key)?.content;
+  return isSectionPath(key) ? curriculumContentFromPath : curriculumContentFromKey;
+}
```

Comment:
Since this could be dealing with curriculum or user/document content, some renaming seems appropriate, e.g. something like:
```suggestion
export function useCurriculumOrDocumentContent(key?: string):  DocumentContentModelType | undefined {
  const curriculumContentFromPath = useCurriculumContentFromPath(key); 
  const documentContentFromKey = useDocumentFromStore(key)?.content;
  return isSectionPath(key) ? curriculumContentFromPath : documentContentFromKey;
}
```